src/apis/youves/utils.ts

```diff
@@ -13,3 +26,88 @@ export const youvesTokensRecord = Object.values(contracts.mainnet)
     }),
     {}
   );
+
+export const fallbackTezosToolkit = new TezosToolkit(getFastRpcClient(MAINNET_SMARTPY_RPC));
+
+class MemoryStorage implements Storage {
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  public storage: Map<string, any>;
+
+  constructor() {
+    this.storage = new Map();
+  }
+  public async set<K extends StorageKey>(key: K, value: StorageKeyReturnType[K]): Promise<void> {
+    this.storage.set(key, value);
+  }
+  public async get<K extends StorageKey>(key: K): Promise<StorageKeyReturnType[K]> {
+    return this.storage.get(key);
+  }
+  public async delete<K extends StorageKey>(key: K): Promise<void> {
+    this.storage.delete(key);
+  }
+  public async clear() {
+    this.storage.clear();
+  }
+}
+
+const getTezosToolkit = (account?: AccountInterface) =>
+  isDefined(account) ? createReadOnlyTezosToolkit(MAINNET_SMARTPY_RPC, account) : fallbackTezosToolkit;
+
+export const createEngineMemoized = memoize(
+  (token: AssetDefinition, account?: AccountInterface) =>
+    createEngine({
+      tezos: getTezosToolkit(account),
+      contracts: token,
+      storage: new MemoryStorage(),
+      indexerConfig: INDEXER_CONFIG,
+      tokens: mainnetTokens,
+      activeCollateral: contracts.mainnet[0].collateralOptions[0],
+      networkConstants: mainnetNetworkConstants
+    }),
+  { cacheKey: args => JSON.stringify(args) }
```

Comment:
Sorry, remembering only the last result will make this memoizing useless because this function is used for multiple values of `token` variable almost simultaneously